go/private/sdk.bzl

```diff
@@ -146,6 +142,13 @@ go_download_sdk_rule = repository_rule(
         "urls": attr.string_list(default = ["https://dl.google.com/go/{}"]),
         "version": attr.string(),
         "strip_prefix": attr.string(default = "go"),
+        "patches": attr.label_list(
+            doc = "A list of patches to apply to the SDK after downloading it",
+        ),
+        "patch_args": attr.string_list(
```

Comment:
I could see an argument for only exposing `patch_strip` here as well. I haven't come across a case where I wanted something else and accidentally falling back to host `patch` is not nice.